src/services/offer.js

```diff
@@ -262,6 +228,59 @@ class OfferService {
             { results };
     }
 
+    /**
+     * Builds a search query with a regular find().
+     * Cannot be used when loading more offers and searching with full-text search at the same time.
+     * Otherwise, use _buildSearchAggregation().
+     */
+    _buildSearchQuery(lastOfferId, value, showAdminReason, showHidden, filters) {
+        let offers;
+        if (lastOfferId) {
+
+            offers = Offer.find({ "$and": [
+                this._buildFilterQuery(filters),
+                { _id: { "$gt": ObjectId(lastOfferId) } }
+            ] });
+
+        } else {
+
+            offers = (value ? Offer.find({ "$and": [
+                this._buildFilterQuery(filters),
+                { "$text": { "$search": value } }
+            ] }, { score: { "$meta": "textScore" } }
+
+            ) : Offer.find(this._buildFilterQuery(filters)));
+        }
+
+        offers.current();
+        if (!showHidden) offers.withoutHidden();
+        if (!showAdminReason) offers.select("-adminReason");
+
+        return offers;
+    }
+
+    /**
+     * Builds a search aggregation with aggregate().
+     * Only use this when loading more offers and searching with full-text search at the same time.
+     * Otherwise, use _buildSearchQuery().
+     */
+    _buildSearchAggregation(lastOfferId, lastOfferScore, value, showAdminReason, showHidden, filters) {
```

Comment:
The function name should not be tied to the implementation, but rather to what it does. Something like `_buildSearchContinuationQuery`. Also update the comment above, whoever reading this want to know what the function does, not what it uses (as they can simply look at L268 for that)